### Life Expectancy and GDP Python Project
#### by Albert Cort Banke
<div>
<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/2/26/World_Health_Organization_Logo.svg/1280px-World_Health_Organization_Logo.svg.png" width="400"/>
</div>

Project scope

**Goal**: To draw insights from the dataset included and illustrate interesting features with visualizations and summary statistics

**Data**: The data stems from and is provided by WHO. Thus the findings will be published at their website

**Analysis**: We use exploratory data analysis and hypothesis testing to assess the data and determine associations in the data. 

1. Preparation

Import the relevant libraries (A) and import and read the data (B)

In [7]:
# (A)
from matplotlib import pyplot as plt
import pandas as pd
import numpy as np
import seaborn as sns

In [27]:
# (B)
df = pd.read_csv('all_data.csv')

# We print the first few observations to check the data has been imported correctly
print(df.head())
# We confirm the data has been imported correctly

  Country  Year  Life expectancy at birth (years)           GDP
0   Chile  2000                              77.3  7.786093e+10
1   Chile  2001                              77.3  7.097992e+10
2   Chile  2002                              77.8  6.973681e+10
3   Chile  2003                              77.9  7.564346e+10
4   Chile  2004                              78.0  9.921039e+10


2. Data transformation

Review (C) the dataset and prepare (D) it for the exploratory data analysis


In [28]:
# (C) To review, we use Pandas built-in functions to gain more information about the data at hand
# The .info() function helps provide this information

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 96 entries, 0 to 95
Data columns (total 4 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Country                           96 non-null     object 
 1   Year                              96 non-null     int64  
 2   Life expectancy at birth (years)  96 non-null     float64
 3   GDP                               96 non-null     float64
dtypes: float64(2), int64(1), object(1)
memory usage: 3.1+ KB
None


Starting from left to right, we see there are 4 columns. These are ***Country***, ***Year***, ***Life expectancy at birth (years)***, and ***GDP***

Moreover there are **96** instances, where 0 non-null values are present. This means we have a complete dataset. 

The data composition is the following: Country = nominal categorical values, Year = quantitative discrete values, Life expectancy at birth (years) = quantitative continuous values, GDP = quantitative continuous values. In short, the variables we are dealing with are both quantitative and categorical. This is important for the following analysis - both exploratory and statistical

During the preparation an aspect of the GDP column became clear. The values are rather large, thus it is difficult to plot the data when the values are too large. Let's inspect the range of the values and confirm the hypothesis.

In [29]:
print(df['GDP'].max())
print(df['GDP'].min())

# Max = 18100000000000.0
# Min = 4415702800.0

# Evidently, the range between the max and min values of the GDP variable is significant
# To remedy this we move onto the next part of the data transformation

18100000000000.0
4415702800.0


In [30]:
# (D) The data transformation will convert the GDP values to be represented in billions (USD)
# The lambda function is used to make a strong and easy implementation of this in the df['GDP'] column

df['GDP'] = df.apply(lambda x: x['GDP']/1000000000, axis = 1)

# Printing the dataframe to inspect if the transformation has been implemented correctly
# As one can see in the GDP column, it has indeed been implemented well

print(df)

     Country  Year  Life expectancy at birth (years)        GDP
0      Chile  2000                              77.3  77.860932
1      Chile  2001                              77.3  70.979924
2      Chile  2002                              77.8  69.736811
3      Chile  2003                              77.9  75.643460
4      Chile  2004                              78.0  99.210393
..       ...   ...                               ...        ...
91  Zimbabwe  2011                              54.9  12.098451
92  Zimbabwe  2012                              56.6  14.242490
93  Zimbabwe  2013                              58.0  15.451769
94  Zimbabwe  2014                              59.2  15.891049
95  Zimbabwe  2015                              60.7  16.304668

[96 rows x 4 columns]


In [32]:
# (D) to ensure that the EDA goes without error we change the column titles. 
# A minor operation which can help when coding the plots

df.columns = ['country', 'year', 'life_expectancy_yrs', 'gdp_usdb']

# Checking whether the column names has been updated 

print(df.columns)

print(df)

# The dataset looks good, and the project continues on the the EDA


Index(['country', 'year', 'life_expectancy_yrs', 'gdp_usdb'], dtype='object')
     country  year  life_expectancy_yrs   gdp_usdb
0      Chile  2000                 77.3  77.860932
1      Chile  2001                 77.3  70.979924
2      Chile  2002                 77.8  69.736811
3      Chile  2003                 77.9  75.643460
4      Chile  2004                 78.0  99.210393
..       ...   ...                  ...        ...
91  Zimbabwe  2011                 54.9  12.098451
92  Zimbabwe  2012                 56.6  14.242490
93  Zimbabwe  2013                 58.0  15.451769
94  Zimbabwe  2014                 59.2  15.891049
95  Zimbabwe  2015                 60.7  16.304668

[96 rows x 4 columns]


3. Exploratory Data Analysis

This section consists of two parts. The first is a summary (E) of the statistical aspects of the data. The second is visualization (F) module, which consists of univariate, bivariate, and multivariate analysis.  

In [33]:
# (E) Summary statistics with Pandas .describe() function

print(df.describe())

              year  life_expectancy_yrs      gdp_usdb
count    96.000000            96.000000     96.000000
mean   2007.500000            72.789583   3880.498571
std       4.633971            10.672882   5197.561311
min    2000.000000            44.300000      4.415703
25%    2003.750000            74.475000    173.301750
50%    2007.500000            76.750000   1280.220000
75%    2011.250000            78.900000   4067.510000
max    2015.000000            81.000000  18100.000000


Interpretation of the statistics

***Year***: The first instance of the data is in year 2000 while the last is in 2015
As this is a time series dataset, the significance of the year attribute is its non-missing status. Furthermore we see the instances are entered in the same way

***Life_expectancy_yrs***: The max expectancy is 81 years while the minimum age is 44.3. This implies that the range in this attribute is significant 
The mean is 72.78 years. In other words, the mean is quite high and close to the max value. 
This could indicate that many of the countries sampled are modernerized

***GDP***: The highest GDP in a year is 18100 billion USD. Meanwhile the lowest in a year is 4.4 billion USD. A strong distance between the two values
The mean GDP is 3880 billion usd. Having such high values implies that the interquartile range will be far from each other. In other words, there will be outliers

Note how the country variable is not represented in the summary table. This is because the variable is categorical and unfit for quantitative analysis